packages/template-retail-react-app/app/pages/product-detail/index.test.js

```diff
@@ -160,3 +169,55 @@ describe('product set', () => {
         })
     })
 })
+
+describe.only('Recommended Products', () => {
+    let fetchMock
+    beforeAll(() => {
+        fetchMock = jest.spyOn(global, 'fetch').mockImplementation(async (url) => {
+            const json = url.endsWith('viewed-recently-einstein') ? einsteinRecommendation : {}
+            return new Response(JSON.stringify(json))
+        })
+    })
+    beforeEach(() => {
+        fetchMock.mockClear()
+    })
+    afterAll(() => {
+        fetchMock.mockRestore()
+    })
+    test('Recently Viewed gets updated when navigating between products', async () => {
+        global.server.use(
+            // Use a single product (and not a product set)
+            rest.get('*/products/:productId', (req, res, ctx) => {
+                return res(ctx.json(productsResponse.data[0]))
+            }),
+            rest.get('*/products', (req, res, ctx) => {
+                return res(ctx.json({}))
+            })
+        )
+        const user = userEvent.setup({
+            advanceTimers: jest.advanceTimersByTime
+        })
+        renderWithProviders(<MockedComponent />)
+
+        let done = false
+        setTimeout(() => (done = true), 150)
+        await waitFor(() => expect(done).toBeTruthy())
```

Comment:
Without this, a _lot_ of errors are logged to console when the `waitFor` does its thing. This is clearly not a good solution, but what's a better one?